### PR TITLE
Offer to install helm into venv

### LINKED ISSUE #608: setup: optionally install helm for user
ask user with inquierer if they want "us" to install helm for them.

if yes: `curl https://raw.githubusercontent.com/helm/helm/main/scripts/get-helm-3 | bash` in a subcommand

if no, show them the URL like current

### PR DESCRIPTION
**Pull request by mplsgrant**
This offers to install helm into the user's venv directory. Fixes #608

I overloaded the `is_helm_installed` function a bit, but I figured that even if we had to do this for every dependency it would remain manageable.

### CODE REVIEW

File: src/warnet/constants.py [hunk 1]
```diff
@@ -98,3 +98,8 @@
     f"helm upgrade --install grafana-dashboards {CHARTS_DIR}/grafana-dashboards --namespace warnet-logging",
     f"helm upgrade --install --namespace warnet-logging loki-grafana grafana/grafana --values {MANIFESTS_DIR}/grafana_values.yaml",
 ]
+
+# Helm binary
+HELM_LATEST_URL = "https://get.helm.sh/helm-latest-version"
```
Comment (pinheadmz) [inline src/warnet/constants.py]: I think we should pin a version and also commit to the expected checksums:
https://github.com/helm/helm/releases

**Comment by pinheadmz** [comment]
` ⭐️ Helm is satisfied: /Users/matthewzipkin/Desktop/work/warnet/.venv/bin/helm`

This is fucking dope dude, great work

**Review by willcl-ark — APPROVED** [review]

**Comment by willcl-ark** [comment]
Yeah this is neat!

Perhaps it would help us to define the install methods we will offer/support (and stick to it!). Something like:

1. Manual: we list the packages you need and link to their install docs. Install them how you wish.
2. Semi-automatic: we offer to fetch bins directly into the local .venv (like here!) for all possible requirements.
3. Fully automatic: a docker image, something like that proposed in #615

Each of the methods should cover **all** required binaries. So for example I'd propose to follow this PR here with a similar `kubectl` and possibley even `k9s` install into venv.

If a user can't/won't install using one of these three ways, then they can't use the tool.

This would hopefully stop this fiasco re-ocurring in the future (or at least, we wouldn't have to worry about it).

File: src/warnet/constants.py (hunk 1 above)
Comment (mplsgrant) [reply src/warnet/constants.py]: I decided to bless v3.16.1 and always pull it: https://github.com/bitcoin-dev-project/warnet/pull/610/commits/07104ee988914e568655dde1e324f752ca6887f8

**Comment by willcl-ark** [comment]
> Reviewers, this pull request conflicts with the following ones:
> 
>     * [#614](https://github.com/bitcoin-dev-project/warnet/pull/614) (swap out kubectl by mplsgrant)

@mplsgrant do you have a preference between which of these goes first? It looks like this one is perhaps more ready?

**Comment by pinheadmz** [comment]
Install looks good but i got this error trying to deploy a network with logging. Not sure if this is an issue with the pinned version of Helm or something wrong with the loki chart in the repo?

warnet error:

`Exception: Error: UPGRADE FAILED: "loki" has no deployed releases`

So i tried to install using helm directly instead of through warnet...


```
$ helm upgrade --install --namespace warnet-logging --create-namespace --values ./resources/manifests/loki_values.yaml loki grafana/loki --version 5.47.2 --dry-run --debug
history.go:56: 2024-09-23 09:43:19.006541243 -0400 EDT m=+0.091649742 [debug] getting history for release loki
upgrade.go:164: 2024-09-23 09:43:19.874007994 -0400 EDT m=+0.959116488 [debug] preparing upgrade for loki
Error: UPGRADE FAILED: "loki" has no deployed releases
helm.go:86: 2024-09-23 09:43:20.456117492 -0400 EDT m=+1.541225945 [debug] "loki" has no deployed releases
UPGRADE FAILED
main.newUpgradeCmd.func2
        helm.sh/helm/v3/cmd/helm/upgrade.go:243
github.com/spf13/cobra.(*Command).execute
        github.com/spf13/cobra@v1.8.1/command.go:985
github.com/spf13/cobra.(*Command).ExecuteC
        github.com/spf13/cobra@v1.8.1/command.go:1117
github.com/spf13/cobra.(*Command).Execute
        github.com/spf13/cobra@v1.8.1/command.go:1041
main.main
        helm.sh/helm/v3/cmd/helm/helm.go:85
runtime.main
        runtime/proc.go:271
runtime.goexit
        runtime/asm_amd64.s:1695
```

**Comment by mplsgrant** [comment]
> Install looks good but i got this error trying to deploy a network with logging. Not sure if this is an issue with the pinned version of Helm or something wrong with the loki chart in the repo?

@pinheadmz I noticed an issue with the architecture mapping. I'm not sure if that had something to do with the error you encountered, but I fixed that and it might be worth trying again.

**Comment by mplsgrant** [comment]
> do you have a preference between which of these goes first? It looks like this one is perhaps more ready?

@willcl-ark I don't have a preference about which goes first. I'm fine to rebase either one.

**Comment by pinheadmz** [comment]
ACK  ec33acea9a40894fbd792bce6a71241132fa2c76
seems to work fine now!

**Comment by mplsgrant** [comment]
Just wanted to remove a bit of dead code before merging.

**Comment by m3dwards** [comment]
On Arm Mac I got: `No Helm binary candidate for arch: None`

Also the question on setup is truncated: 

```shell
[?] Would you like to use Warnet's downloader to install Helm into your virtual environme...:
```

**Comment by pinheadmz** [comment]
Can confirm arm macos issue, but the message printed fine for me

Screenshot 2024-09-27 at 10 08 10 AM

**Comment by m3dwards** [comment]
The truncation must be because I run narrow terminals. Perhaps the question could be a bit shorter? Assuming ~~click~~ inquirer doesn't have a wrap text option.

**Comment by mplsgrant** [comment]
> On Arm Mac I got: `No Helm binary candidate for arch: None`

@m3dwards What happens when you run this in a python terminal?

````python
import os
os.uname().machine
````

**Comment by pinheadmz** [comment]
```
>>> import os
>>> os.uname()
posix.uname_result(sysname='Darwin', nodename=' ', release='23.5.0', version='Darwin Kernel Version 23.5.0: Wed May  1 20:16:51 PDT 2024; root:xnu-10063.121.3~5/RELEASE_ARM64_T8103', machine='arm64')
>>> os.uname().machine
'arm64'

```

**Comment by m3dwards** [comment]
> arm64

**Comment by m3dwards** [comment]
Works! Great PR!

**Comment by pinheadmz** [comment]
post merge ACK,
tested on macos arm64 and debian x86_64

**Pull request merged**